doc/external-libs/ginkgo.html

```diff
@@ -0,0 +1,98 @@
+        highly optimized fine grid level linear algebra operations. It currently has
+        the capability to solve on the CPU with the support of OpenMP and on the GPU
+        with NVIDIA's cuda libraries. For some operations it uses NVIDIA's own libraries, CuSparse
+        and CuBLAS but some routines also have self implemented cuda functions. See the documentation
+        for more details.
+    </p>
+
+    <p>
+      <acronym>deal.II</acronym> has wrapper classes to the linear algebra
+      parts of Ginkgo that provide almost the
+      same interfaces as the built-in <acronym>deal.II</acronym> linear algebra operations.
+    </p>
+
+    <h2>Installing <acronym>deal.II</acronym> with Ginkgo</h2>
+
+    <p>
+        During the cmake configuration, the following flags should be specified
+        <code>-DDEAL_II_WITH_GINKGO=ON</code>, to enable Ginkgo and the <code>-DGINKGO_DIR=DIR</code>
+        can be set with the path where Ginkgo has been installed. 
+    </p>
+
+    <h3>Installing Ginkgo</h3>
+
+    <p>
+        Installing Ginkgo is quite simple. Ginkgo currently uses Cmake and hence one can use
+        the following commands to easily install Ginkgo. For different dependencies and compatible
+        libraries, please refer to <a href="https://github.com/ginkgo-project/ginkgo"
+                                      target="_top">Ginkgo</a>.
+        The different flags that can be added are:
+        <code> -DBUILD_REFERENCE={ON,OFF}:</code> Builds the reference single thread implementation
+        used to check the more sophisticated implementations of the GPU/CPU.
+        <code> -DBUILD_CUDA={ON,OFF}:</code> Builds the GPU implementation specifically the NVIDIA
+        cuda implementations. This needs CUDA to be installed on the machine.
+        <code> -DBUILD_OMP={ON,OFF}:</code> Builds the multi-threaded OpenMP implementations.
+
+      <pre>
+
+	git clone https://github.com/ginkgo-project/ginkgo.git
+  mkdir build; cd build 
+  cmake -DBUILD_REFERENCE=on/off -DBUILD_CUDA=on/off -DBUILD_OMP=on/off -DCMAKE_INSTALL_PREFIX=/path/to/install/ -DCMAKE_BUILD_TYPE=Debug
+	make install
+      </pre>
+    </p>
+
+    <h4>Running Ginkgo tests</h4>
+
+    <p>
+        Ginkgo uses <code>gtests</code> to run unit tests. The reference implementations are
+        non-optimized versions to which the OpenMP and cuda versions are compared with for
+        correctness.
+      <pre>
+
+	make test
+      </pre>
+      The output should contain several lines of the form:
+      <pre>
+     Start  1: path/to/test
+ 1/13 Test  #1: path/to/test .............................   Passed    0.01 sec
+      </pre>
+      To run only a specific test for more details, from the build directory, run
+      <pre>
+	./path/to/test
+      </pre>
+    </p>
+
+    <h5>NOTES </h5>
+    <p>
+        Currently, Ginkgo needs to be compiled in "debug mode", due to some problems with
+        a dependent <code>gflags</code> dependency. 
```

Comment:
> dependent... dependency

This is quite awkward. Could you try to rephrase this to make more clear what it means?